src/commands/test.js

```diff
@@ -11,6 +14,10 @@ const test = (context) => {
     extraEnv.DETAILED_LOG_TO_STDOUT = 'true';
   }
 
+  if (process.version !== LAMBDA_VERSION) {
+    throw new Error(`You're running tests on Node ${process.version}, but Zapier runs your code on ${LAMBDA_VERSION}. To ensure errors don't sneak up on you, the version numbers must match.`);
```

Comment:
I don't love pushing NVM as a requirement since it's not (just a suggestion to manage versions). I'd rather have something in the readme that goes into the version requirements and give NVM as a suggestion (along with [n](https://github.com/tj/n) and any other good tools). 

The error message should specify what the error is (version mismatch) and then could point to the docs for resolution suggestions). 

How about `You're running tests on Node ${process.version}, but Zapier runs your code on ${LAMBDA_VERSION}. Your test version must match. See ${README_LINK} for suggestions.`